sign/dilithium/mode3/internal/dilithium_test.go

```diff
@@ -0,0 +1,107 @@
+package internal
+
+import (
+	"encoding/binary"
+	"testing"
+
+	common "github.com/cloudflare/circl/sign/dilithium/internal"
+)
+
+// Checks whether p is normalized.  Only used in tests.
+func PolyNormalized(p *common.Poly) bool {
+	p2 := *p
+	p2.Normalize()
+	return p2 == *p
+}
+
+func BenchmarkVerify(b *testing.B) {
+	var seed [32]byte
+	var msg [8]byte
+	var sig [SignatureSize]byte
+	pk, sk := NewKeyFromSeed(&seed)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		binary.LittleEndian.PutUint64(msg[:], uint64(i))
+		SignTo(sk, msg[:], sig[:])
+		b.StartTimer()
+		Verify(pk, msg[:], sig[:])
+	}
+}
+
+func BenchmarkSign(b *testing.B) {
+	var seed [32]byte
+	var msg [8]byte
+	var sig [SignatureSize]byte
+	_, sk := NewKeyFromSeed(&seed)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		binary.LittleEndian.PutUint64(msg[:], uint64(i))
+		SignTo(sk, msg[:], sig[:])
+	}
+}
+
+func BenchmarkGenerateKey(b *testing.B) {
+	var seed [32]byte
+	for i := 0; i < b.N; i++ {
+		NewKeyFromSeed(&seed)
+	}
+}
+
+func TestSign(t *testing.T) {
+	var seed [common.SeedSize]byte
+	var sig [SignatureSize]byte
+	msg := []byte{
```

Comment:
This test should verify random inputs.

Reply:
The test `TestSignThenVerifyAndPkSkPacking` does that (and more).  I keep this test with very specific inputs to compare it easily with the reference implementation.

Reply:
Hence, provide the source of this vector.

Reply:
I removed the test as it was redundant (superseded by `TestSignThenVerifyAndPkSkPacking`.)